bzl/tbb.BUILD

```diff
@@ -1,67 +1,168 @@
+    hdrs = [
+        ":gen_version_file",
+    ],
+    defines = [
+        "TBB_USE_THREADING_TOOLS",
+        "__TBB_DYNAMIC_LOAD_ENABLED=0",
+        "__TBB_SOURCE_DIRECTLY_INCLUDED=1",
+        "__TBB_x86_64=1",
+    ] + select({
+        "@bazel_tools//src/conditions:windows": [
+            "USE_WINTHREAD",
+            "__TBB_CPU_CTL_ENV_PRESENT=1",
+        ],
+        "//conditions:default": [
+            "USE_PTHREAD",
+        ],
+    }),
+    includes = [
+        "build/vs2013",
+        "include",
+        "src",
+        "src/rml/include",
+    ],
 )
 
-### TODO: This is only for testing the parallel execution.
-### Delete this when the linking of TBB is fully integrated
-### to be built natively with Bazel.
-# genrule(
-#   name = "build_tbb",
-#   srcs = glob(["**"]) + [
-#     "@local_config_cc//:toolchain",
-#   ],
-#   cmd = """
-#         # set -e
-#         WORK_DIR=$$PWD
-#         DEST_DIR=$$PWD/$(@D)
-#
-#         cd $$(dirname $(location :Makefile))
-#
-#          #TBB's build needs some help to figure out what compiler it's using
-#          #if $$CXX --version | grep clang &> /dev/null; then
-#         COMPILER_OPT="compiler=clang"
-#          #else
-#          #  COMPILER_OPT="compiler=gcc"
-#          #fi
-#
-#          # uses extra_inc=big_iron.inc to specify that static libraries are
-#          # built. See https://software.intel.com/en-us/forums/intel-threading-building-blocks/topic/297792
-#         make tbb_build_prefix="build" \
-#             extra_inc=big_iron.inc \
-#             $$COMPILER_OPT; \
-#
-#         echo cp build/build_{release,debug}/*.a $$DEST_DIR
-#         cp build/build_{release,debug}/*.a $$DEST_DIR
-#         cd $$WORK_DIR
-#   """,
-#   outs = [
-#     "libtbb.a",
-#     "libtbbmalloc.a",
-#   ]
-# )
-#
-# cc_library(
-#     name = "tbb",
-#     hdrs = glob([
-#         "include/serial/**",
-#         "include/tbb/**/**",
-#         ]),
-#     srcs = ["libtbb.a"],
-#     includes = ["include"],
-#     visibility = ["//visibility:public"],
-# )
+genrule(
+    name = "tbb_unix",
+    srcs = glob(["**"]),
+    outs = [
+        "libtbb.a",
+        "libtbbmalloc.a",
+    ],
+    cmd = """
+        # set -e
+        WORK_DIR=$$PWD
+        DEST_DIR=$$PWD/$(@D)
+        cd $$(dirname $(location :Makefile))
+        #TBB's build needs some help to figure out what compiler it's using
+        if $$CXX --version | grep clang &> /dev/null; then
+          COMPILER_OPT="compiler=clang"
+        else
+          COMPILER_OPT="compiler=gcc"
+        fi
+
+        # uses extra_inc=big_iron.inc to specify that static libraries are
+        # built. See https://software.intel.com/en-us/forums/intel-threading-building-blocks/topic/297792
+        make tbb_build_prefix="build" \
+            extra_inc=big_iron.inc \
+            $$COMPILER_OPT; \
+
+        echo cp build/build_{release,debug}/*.a $$DEST_DIR
+        cp build/build_{release,debug}/*.a $$DEST_DIR
+        cd $$WORK_DIR
+  """,
```

Comment:
If we're going to use make to do the build, we need to ensure that it's using the internal linux toolchain that is based on GLIBC 2.17 (so that we can target ancient linux distros like CentOS).

Reply:
How do I specify that?

Reply:
The way I know how to do it is to use bazel to do the build, not rely on `make`. But perhaps there's a way to get the right env vars passed into `make`.

I thought we had it building for linux/macos under bazel before?